core/src/main/scala/org/apache/spark/sql/rapids/tool/util/FSUtils.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.tool.util
+
+import java.io.{BufferedWriter, FileOutputStream, OutputStreamWriter}
+import java.nio.charset.StandardCharsets
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, FSDataOutputStream, LocalFileSystem, Path}
+import org.apache.hadoop.fs.permission.FsPermission
+
+/**
+ * Utility functions to interact with the file system (HDFS, local, etc).
+ */
+object FSUtils {
+  // use same as Spark event log writer
+  private val LOG_FILE_PERMISSIONS = new FsPermission(Integer.parseInt("660", 8).toShort)
+  private val LOG_FOLDER_PERMISSIONS = new FsPermission(Integer.parseInt("770", 8).toShort)
+
+  // Copied from org.apache.spark.streaming.util.HdfsUtils
+  def getFSForPath(path: Path, hadoopConf: Configuration): FileSystem = {
+    // For local file systems, return the raw local file system, such calls to flush()
+    // actually flushes the stream.
+    val fs = path.getFileSystem(hadoopConf)
+    fs match {
+      case localFs: LocalFileSystem => localFs.getRawFileSystem
+      case _ => fs
+    }
+  }
+
+  def getOutputStream(outputPath: String, hadoopConf: Configuration): FSDataOutputStream = {
+    val dfsPath = new Path(outputPath)
+    getOutputStream(dfsPath, hadoopConf)
+  }
+
+  def getOutputStream(dfsPath: Path, hadoopConf: Configuration): FSDataOutputStream = {
```

Comment:
if we don't want people to use this make it private

Reply:
I changed them to private.